gubled/gubled_test.go

```diff
@@ -140,7 +140,13 @@ func TestGcmOnlyStartedIfEnabled(t *testing.T) {
 	defer mockCtrl.Finish()
 
 	routerMock := NewMockPubSubSource(mockCtrl)
+
+	amMock := NewMockAccessManager(mockCtrl)
+	msMock := NewMockMessageStore(mockCtrl)
+
 	routerMock.EXPECT().KVStore().Return(store.NewMemoryKVStore(), nil)
+	routerMock.EXPECT().AccessManager().Return(amMock, nil).MaxTimes(2)
+	routerMock.EXPECT().MessageStore().Return(msMock, nil).MaxTimes(2)
```

Comment:
There is a problem with this. Here I'm testing something in `gubled` using a mock from server (`RouterMock`). Re your comment above this means I should either:

* create a `RouterMock` inside the `gubled` package in a `_test.go` file and it will be visible only here.
* create a public `RouterMock` in the `server` package and use it in other packages.